src/main/java/command/AddCommand.java

```diff
@@ -0,0 +1,102 @@
+package command;
+
+import tasks.DeadLine;
+import tasks.Event;
+import tasks.ToDo;
+import utils.Storage;
+import utils.TaskList;
+import utils.Ui;
+
+import java.io.IOException;
+
+/**
+ * AddCommand is a Command, specifically for todo, event and deadlines.
+ */
+public class AddCommand extends Command {
+
+    private String type;
+    private String taskDescription;
+    private String time;
+
+    /**
+     * Constructor for AddCommand, for type of: event or deadlines.
+     */
+    public AddCommand(String type, String taskDescription, String time) {
+        this.type = type;
+        this.taskDescription = taskDescription;
+        this.time = time;
+    }
+
+    /**
+     * Constructor for AddCommand, for type of: Todo.
+     */
+    public AddCommand(String type, String taskDescription) {
+        this.type = type;
+        this.taskDescription = taskDescription;
+    }
+
+    /**
+     * Executes the given task and prints the respective output.
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+
+        String output = "";
+
+        switch (type) {
+        case "todo": {
+            // Create ToDo object
+            ToDo t = new ToDo(taskDescription);
+            tasks.addTask(t);
+
+            // Saving to text file
```

Comment:
I really like how there are comments within the methods themselves instead of just dumping everything in the javadoc comment up top. Makes the code easier to understand and aids in debugging.